protex/update.py

```diff
@@ -102,7 +104,7 @@ def _update(self, candidates: list[tuple], nr_of_steps: int) -> None:
         pass
 
     def _adapt_probabilities(
-        self, to_adapt=list[tuple[str, int, frozenset[str]]]
+        self, to_adapt=list[tuple[str, int, frozenset[str]]], K=int
```

Comment:
Do not add K to the function signature here. It is already accessible inside the class (as defined in line 65) via self.K.